starter-core/src/main/resources/pom.xml

```diff
@@ -24,7 +24,7 @@
         <dependency>
             <groupId>io.kotest</groupId>
             <artifactId>kotest-runner-junit5-jvm</artifactId>
-            <version>5.6.0</version>
+            <version>4.2.5</version>
```

Comment:
why this?

Reply:
Because otherwise

```
 [ERROR] /home/runner/.m2/repository/io/kotest/kotest-extensions-jvm/5.6.0/kotest-extensions-jvm-5.6.0.jar!/META-INF/kotest-extensions.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.	
  [ERROR] /home/runner/.m2/repository/io/kotest/kotest-framework-api-jvm/5.6.0/kotest-framework-api-jvm-5.6.0.jar!/META-INF/kotest-framework-api.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.	
  [ERROR] /home/runner/.m2/repository/io/kotest/kotest-framework-concurrency-jvm/5.6.0/kotest-framework-concurrency-jvm-5.6.0.jar!/META-INF/kotest-framework-concurrency.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.	
  [ERROR] /home/runner/.m2/repository/io/kotest/kotest-framework-discovery-jvm/5.6.0/kotest-framework-discovery-jvm-5.6.0.jar!/META-INF/kotest-framework-discovery.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.	
  [ERROR] /home/runner/.m2/repository/io/kotest/kotest-framework-engine-jvm/5.6.0/kotest-framework-engine-jvm-5.6.0.jar!/META-INF/kotest-framework-engine.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.	
  [ERROR] /home/runner/.m2/repository/io/kotest/kotest-runner-junit5-jvm/5.6.0/kotest-runner-junit5-jvm-5.6.0.jar!/META-INF/kotest-runner-junit5.kotlin_module: (-1, -1) Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.	
```

https://github.com/micronaut-projects/micronaut-starter/actions/runs/6311236421/job/17134842467
https://ge.micronaut.io/s/iebbnlrfxh42k/console-log/task/:test-cli:test?page=1

I don't believe this merge (with a major update) was the right thing to do?

https://github.com/micronaut-projects/micronaut-starter/pull/1713

Reply:
The version is 4.2.5 on the 3.9.x, 3.10.x, 4.0.x and 4.1.x branches

I think it should have been merged for 4.0.0, not 3.8.x 🤔

Reply:
Actually, I'm not sure if the dependencies

```
kotest-assertions-core-jvm = { module = "io.kotest:kotest-assertions-core-jvm", version.ref = "kotest" }
kotest-runner-junit5-jvm = { module = "io.kotest:kotest-runner-junit5-jvm", version.ref = "kotest" }
```

Are used in Micronaut 4 🤔

Reply:
in Micronaut Framework 4, we only offer KoTest 5. 

In Micronaut Framework 3 we offered by default KoTest 4 but users could opt-in to KoTest 5.